File: Leetcode/3438. Find Valid Pair of Adjacent Digits in String.py
class Solution:
    def findValidPair(self, s: str) -> str:
        f=Counter(s)
        for i in range(len(s)-1):
            if s[i]!=s[i+1] and f[s[i]]==int(s[i]) and f[s[i+1]]==int(s[i+1]):
                return s[i:i+2]
        return ""

File: Leetcode/3438. Find Valid Pair of Adjacent Digits in String.cpp
class Solution {
public:
    string findValidPair(string s) {
        unordered_map<char,int> m;
        int n=s.size();
        for(int i=0;i<n;i++){
            m[s[i]]++;
        }
        for(int i=0;i<n-1;i++){
            if(s[i]!=s[i+1] && (m[s[i]]==(s[i]-'0')) && (m[s[i+1]]==(s[i+1]-'0'))){
                return s.substr(i,2);
            }
        }
        return "";
    }
};
